File: exercises/python_playwright/overlapped_element/overlapped_element_playwright.py
from playwright.sync_api import Page


class OverlappedElementPage:
    def __init__(self, page: Page):
        """Page object for the Overlapped Element exercise."""
        self.page = page
        self.name_field = page.get_by_placeholder("Name")

    def open(self) -> None:
        """Open the Overlapped Element exercise page."""
        self.page.goto("http://uitestingplayground.com/overlapped")

    def input_name(self, text: str) -> None:
        """Scroll the name field into view and fill it."""
        field = self.name_field
        field.hover()
        field.evaluate("element => element.scrollTop += 100")
        field.fill(text)

File: exercises/javascript_playwright/overlapped-element/overlappedElementPlaywright.js
export class OverlappedElementPage {
    /**
     *
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.nameInput = page.locator("#name");
    }

    async openExercisePage() {
        await this.page.goto("http://uitestingplayground.com/overlapped");
    }

    async scrollToNameInput() {
        await this.nameInput.click();
        await this.nameInput.evaluate((element) => (element.scrollTop += 100));
    }

    async fillNameInput(name) {
        await this.nameInput.fill(name);
    }

    async readNameInputValue() {
        return await this.nameInput.inputValue();
    }
}
